### ISSUE TITLE
Functools module

### ISSUE THREAD

**Issue by WillDaSilva** [dictu-lang/Dictu#424]
It'd be nice to have a module (written in Dictu) for higher-order functions. Some examples of functions that could be included:
- partial
- memoize

More could be added, especially if support for generic collection iteration is added, but this would be a nice start.

**Comment by WillDaSilva** [comment]
@Jason2605 do you have any suggestions for how a module written in Dictu could be added to the standard library?

**Comment by Jason2605** [comment]
@WillDaSilva Yep, we went through it in this issue here: https://github.com/dictu-lang/Dictu/issues/431

If none of that makes sense happy to go through it again though. Probably worth me making some documentation on this
